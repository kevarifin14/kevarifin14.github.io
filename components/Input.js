export default function Input({
  placeholder, type, required, value, onChange,
}) {
  return (
    <input
      className="outline-none sm:text-xl text-md col-span-1 mr-2 rounded-none border-b border-solid bg-transparent border-black"
      required={required}
      type={type}
      placeholder={placeholder}
      onChange={onChange}
      value={value}
    />
  );
}
