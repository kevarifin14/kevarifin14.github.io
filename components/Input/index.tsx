import { forwardRef, HTMLProps } from 'react';

import { classNames } from 'utils/tailwind';

interface InputProps extends HTMLProps<HTMLInputElement>{}

const Input = forwardRef<HTMLInputElement, InputProps>(({ size, ...inputProps }, ref) => {
  const inputClassName = classNames(
    'focus:outline-none', 'w-full', 'border', 'bg-light',
    'rounded-md', 'block', 'text-gray-900', 'placeholder-gray-500', 'focus:ring-1',
    'focus:border-primary', 'focus:ring-primary',
    'px-5', 'py-3',
    inputProps.className,
  );

  return (
    <div className="w-full">
      <div className="relative">
        <input
          {...inputProps}
          className={inputClassName}
          ref={ref}
          style={{ WebkitAppearance: 'none' }}
        />
      </div>
    </div>
  );
});

export default Input;
