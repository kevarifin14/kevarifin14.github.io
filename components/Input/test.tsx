import renderer from 'react-test-renderer';

import Input from '.';

describe('Input', () => {
  it('matches snapshot', () => {
    const tree = renderer.create(
      <Input />,
    );
    expect(tree).toMatchSnapshot();
  });
});
