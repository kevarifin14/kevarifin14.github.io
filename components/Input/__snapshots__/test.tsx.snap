// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input matches snapshot 1`] = `
<div
  className="w-full"
>
  <div
    className="relative"
  >
    <input
      className="shadow-sm focus:outline-none w-full border bg-light border-gray-300 rounded-md block text-gray-900 placeholder-gray-500 focus:border-primary focus:ring-primary"
      style={
        Object {
          "WebkitAppearance": "none",
        }
      }
    />
  </div>
</div>
`;
